src/avram/where.cr

```diff
@@ -1,31 +1,52 @@
 module Avram::Where
-  abstract class SqlClause
-    getter :column
-    getter :value
+  abstract class Condition
+    abstract def prepare(placeholder_supplier : Proc(String)) : String
 
-    def initialize(@column : Symbol | String, @value : String | Array(String) | Array(Int32))
+    def ==(other : Condition)
+      prepare(->{ "unused" }) == other.prepare(->{ "unused" })
     end
 
-    abstract def operator : String
-    abstract def negated : SqlClause
-
-    def prepare(prepared_statement_placeholder : String)
-      "#{column} #{operator} #{prepared_statement_placeholder}"
+    def ==(other)
+      false
     end
 
     def clone
       self
     end
   end
 
-  abstract class NullSqlClause < SqlClause
-    @value = "NULL"
+  abstract class SqlClause < Condition
+    getter column : Symbol | String
 
-    def initialize(@column : Symbol | String)
+    def initialize(@column)
     end
 
-    def prepare
-      "#{column} #{operator} #{@value}"
+    abstract def operator : String
+    abstract def negated : SqlClause
+
+    def prepare(placeholder_supplier : Proc(String)) : String
+      "#{column} #{operator} #{placeholder_supplier.call}"
+    end
+  end
+
+  abstract class ValueHoldingSqlClause < SqlClause
+    getter value : String | Array(String) | Array(Int32)
+
+    def initialize(@column, @value)
+    end
+
+    def ==(other : ValueHoldingSqlClause)
+      (prepare(->{ "unused" }) + value.to_s) == (other.prepare(->{ "unused" }) + other.value.to_s)
+    end
+
+    def ==(other : Condition)
+      false
+    end
+  end
+
+  abstract class NullSqlClause < SqlClause
+    def prepare(_placeholder_supplier : Proc(String))
```

Comment:
```suggestion
    def prepare(_placeholder_supplier : Proc(String)) : String
```

This should probably have the return type to keep it consistent with the abstract.